toptal/haste-client
# [UTI-2583] Adapt haste-client to the new API usage rules

[UTI-2583](https://toptal-core.atlassian.net/browse/UTI-2583)

### Description

- Added share url andserver token environment variables
- Adapted client to new hastebin api
- Updated readme 
- Added licence file
- Added pr template
- (hastebin.com values in code and documentation will be updated later before merge )

### How to test

- set environment variables
- `bundle install`
- Create a text or code file
- `./bin/haste filename` ex: `./bin/haste README.md`
- It will generate share page url
- Check share page if it is working
- Do same steps with  `cat filename | ./bin/haste`
- Do same steps with  `cat filename | ./bin/haste | pbcopy` (It will paste url to clipboard)

### Environment variables

 You should add the following to your ~.bash_profile:

```bash
export HASTE_SERVER_TOKEN="yourtoken"
export HASTE_SERVER="https://new.hastebin.com"
export HASTE_SHARE_SERVER="https://new.hastebin.com"
```

### Acceptance Criteria

- [ ] The new version of hastebin works correctly with haste-client.
- [ ] We leave the PR ready to be merged so we can sync it with Hastebin publication.
- [ ] We have the documentation of the project up to date.### Pre-merge checklist

### Pre-merge checklist

- [ ] A snapshot of the author's env vars has been added


[UTI-2583]: https://toptal-core.atlassian.net/browse/UTI-2583?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
